### PR TITLE
Continuation of #116: use the proper type for objects when c:type is gpointer

### PR DESCRIPTION
**Pull request by gkoz**
A case handled by this PR is
```xml
<type name="StatusIcon" c:type="gpointer"/>
```

### CODE REVIEW

File: src/codegen/sys/ffi_type.rs [hunk 1]
```diff
@@ -15,12 +15,21 @@ pub fn ffi_type(env: &Env, tid: library::TypeId, c_type: &str) -> Result {
         if let Some(c_tid) = env.library.find_type(0, c_type) {
             // Fast track plain fundamental types avoiding some checks
             if env.library.type_(c_tid).maybe_ref_as::<Fundamental>().is_some() {
```
Comment (EPashkin) [inline src/codegen/sys/ffi_type.rs]: IMHO this not allow Classes in.

Comment (gkoz) [inline src/codegen/sys/ffi_type.rs]: This tests up if `c_type` names a fundamental.
Updated the PR descr.

**Comment by gkoz** [comment]
I admit this `ffi_type` has become convoluted, confusing and somewhat redundant. I've already started rewriting it but that is what prompted the desire to alter the architecture.

**Comment by EPashkin** [comment]
Ok. Thanks.

**Comment by gkoz** [comment]
@homu r+

**Pull request merged**